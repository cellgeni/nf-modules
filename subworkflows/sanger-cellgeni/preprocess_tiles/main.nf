include { IMAGING_PREPROCESS } from "../../../modules/sanger-cellgeni/imaging/preprocess"
include { IMAGING_GENERATECOMPANION } from "../../../modules/sanger-cellgeni/imaging/generatecompanion"

workflow PREPROCESS_TILES {
    take:
    master_file_ch // channel: [ val(meta), val(round_index), file(master_file_folder), val(master_file_name) ]
    psf_folder

    main:
    ch_versions = Channel.empty()

    IMAGING_GENERATECOMPANION(master_file_ch)
    ch_versions = ch_versions.mix(IMAGING_GENERATECOMPANION.out.versions.first())

    tiles = IMAGING_GENERATECOMPANION.out.csv
        .splitCsv(header: true, strip: true, sep: ",")
        .map { row ->
            [row[0], file(row[1].root_xml), row[1].master_file, row[1].image_id, row[1].index]
        }
    IMAGING_PREPROCESS(tiles, psf_folder)
    ch_versions = ch_versions.mix(IMAGING_PREPROCESS.out.versions.first())

    emit:
    companion_tiles = IMAGING_GENERATECOMPANION.out.companion.combine(IMAGING_PREPROCESS.out.fovs, by: 0).groupTuple(by: [0, 1]) // [ meta, companion, [fovs] ]
    versions = ch_versions // channel: [ versions.yml ]
}
