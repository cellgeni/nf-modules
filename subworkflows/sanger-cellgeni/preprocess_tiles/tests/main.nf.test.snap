{
    "preprocess_tiles - xml - decon_10": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "xml_decon"
                        },
                        "xml_decon_image0.companion.ome:md5,c228d074eba208f27e16c33d7192c994",
                        [
                            "xml_decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ],
                "companion_tiles": [
                    [
                        {
                            "id": "xml_decon"
                        },
                        "xml_decon_image0.companion.ome:md5,c228d074eba208f27e16c33d7192c994",
                        [
                            "xml_decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-17T16:22:39.978608949"
    },
    "preprocess_tiles - xml - no_decon": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "xml_no-decon"
                        },
                        "xml_no-decon_image0.companion.ome:md5,2449ae4f2fb30495364ea5da0f00e09f",
                        [
                            "xml_no-decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ],
                "companion_tiles": [
                    [
                        {
                            "id": "xml_no-decon"
                        },
                        "xml_no-decon_image0.companion.ome:md5,2449ae4f2fb30495364ea5da0f00e09f",
                        [
                            "xml_no-decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-17T16:21:18.568400573"
    }
}